### PR TITLE
cherrypy 18.9.0

### PR DESCRIPTION
**Pull request by SC426**
**Destination channel:** defaults

### Links

- [PKG-4672](https://anaconda.atlassian.net/browse/PKG-4672)
- [Upstream repository](https://github.com/cherrypy/cherrypy/tree/v18.9.0)
- [Upstream changelog/diff](https://github.com/cherrypy/cherrypy/blob/v18.9.0/CHANGES.rst)

### Explanation of changes:

- Update version
- Move extra packages to `run_constrained`

[PKG-4672]: https://anaconda.atlassian.net/browse/PKG-4672?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,15 +23,16 @@ requirements:
     - wheel
   run:
     - python
-    - more-itertools
     - cheroot >=8.2.1
     - portend >=2.1.1
+    - more-itertools
     - zc.lockfile
     - jaraco.collections
+    - pywin32 >=227   # [win and py<310]
```
Comment (JamesRobertsonGames) [inline recipe/meta.yaml:31]: Add importlib-metadata for low versions
https://github.com/cherrypy/cherrypy/blob/main/setup.py#L73C10-L73C28
```suggestion
    - pywin32 >=227   # [win and py<310]
    - importlib-metadata # [py<38]
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -23,15 +23,16 @@ requirements:
     - wheel
```
Comment (JamesRobertsonGames) [inline recipe/meta.yaml:23]: I think that setuptools is necessary for the build system
https://github.com/cherrypy/cherrypy/blob/89461873c230ac853da6154a4ec1aaaa723f938e/pyproject.toml#L4

```suggestion
    - wheel
    - setuptools >=45
```

**Review by JamesRobertsonGames — COMMENTED** [review]
There may also be a need add pinning for 
```
setuptools_scm[toml] >= 3.5"
```
https://github.com/cherrypy/cherrypy/blob/89461873c230ac853da6154a4ec1aaaa723f938e/pyproject.toml#L7C1-L8C1

**Review by boldorider4 — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:25]: It's already three lines above

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:33]: It's from the main branch. For v18.9.0, it's not needed https://github.com/cherrypy/cherrypy/blob/v18.9.0/setup.py#L68-L72

**Review by cbouss — APPROVED** [review]

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**